### PR TITLE
[flink] CDC Ingestion supported metadata columns

### LINKED ISSUE #1985: [Feature]  CDC Ingestion supported metadata columns
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

CDC Ingestion supported metadata columns
图片

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

close #1985

### Tests

MySqlSyncDatabaseActionITCase#testMetadataColumns()
MySqlSyncTableActionITCase#testMetadataColumns()

### CODE REVIEW

**Comment by JingsongLi** [comment]
Please rebase master

**Comment by MonsterChenzhuo** [comment]
@JingsongLi PTAL,thanks.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    OP_TS(
+            "op_ts",
+            DataTypes.TIMESTAMP(3).notNull(),
+            new CdcMetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Map<String, String> read(JsonNode record) {
+                    String dbName =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:110]: dbName?

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    /** Name of the table that contain the row. */
+    TABLE_NAME(
+            "table_name",
+            DataTypes.STRING().notNull(),
+            new CdcMetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Map<String, String> read(JsonNode record) {
+                    String dbName =
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:53]: dbName?

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcMetadataConverter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * A functional interface for converting CDC metadata.
+ *
+ * <p>This interface provides a mechanism to convert Change Data Capture (CDC) metadata from a given
+ * {@link JsonNode} source. Implementations of this interface can be used to process and transform
+ * metadata entries from CDC sources.
+ */
+public interface CdcMetadataConverter extends Serializable {
+
+    Map<String, String> read(JsonNode var1);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcMetadataConverter.java:37]: var1 to a meaningful name

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java [hunk 4]
```diff
@@ -416,6 +425,10 @@ else if (Date.SCHEMA_NAME.equals(className)) {
                     computedColumn.eval(resultMap.get(computedColumn.fieldReference())));
         }
 
+        for (CdcMetadataConverter metadataConverters : metadataConverters) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:428]: `metadataConverter`

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                            record.get("source").get(AbstractSourceInfo.TABLE_NAME_KEY).asText();
+                    Map<String, String> resultMap = new HashMap<>();
+                    resultMap.put("table_name", dbName);
+                    return resultMap;
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:57]: `Collections.singletonMap`

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            new CdcMetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Map<String, String> read(JsonNode record) {
+                    String dbName =
+                            record.get("source").get(AbstractSourceInfo.DATABASE_NAME_KEY).asText();
+                    Map<String, String> resultMap = new HashMap<>();
+                    resultMap.put("database_name", dbName);
+                    return resultMap;
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:84]: `Collections.singletonMap`

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+            new CdcMetadataConverter() {
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Map<String, String> read(JsonNode record) {
+                    String dbName =
+                            record.get("source").get(AbstractSourceInfo.TIMESTAMP_KEY).asText();
+                    Map<String, String> resultMap = new HashMap<>();
+                    resultMap.put("op_ts", dbName);
+                    return resultMap;
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:114]: `Collections.singletonMap`

**Review by JingsongLi — CHANGES_REQUESTED** [review]
Thanks @MonsterChenzhuo , left some minor comments.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:110]: 图片

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 7 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:114]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 5 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:57]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java (hunk 4 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlDebeziumJsonEventParser.java:428]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcMetadataConverter.java (hunk 3 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcMetadataConverter.java:37]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 2 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:53]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java:110]: done.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java (hunk 6 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlMetadataProcessor.java]: done.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**